components/service/glean/src/test/java/mozilla/components/service/glean/GleanTest.kt

```diff
@@ -513,4 +501,112 @@ class GleanTest {
         assertEquals(1, pingStringMetrics.length())
         assertEquals(testValue, pingStringMetrics.get("telemetry.string_metric"))
     }
+
+    @Test
+    fun `Basic metrics should be cleared when disabling uploading`() {
+        val stringMetric = StringMetricType(
+            disabled = false,
+            category = "telemetry",
+            lifetime = Lifetime.Ping,
+            name = "string_metric",
+            sendInPings = listOf("default")
+        )
+
+        stringMetric.set("TEST VALUE")
+        assertTrue(stringMetric.testHasValue())
+
+        Glean.setUploadEnabled(false)
+        assertFalse(stringMetric.testHasValue())
+        stringMetric.set("TEST VALUE")
+        assertFalse(stringMetric.testHasValue())
+
+        Glean.setUploadEnabled(true)
+        assertFalse(stringMetric.testHasValue())
+        stringMetric.set("TEST VALUE")
+        assertTrue(stringMetric.testHasValue())
+    }
+
+    @Test
+    fun `Core metrics should be cleared and restored when disabling and enabling uploading`() {
+        assertTrue(GleanInternalMetrics.os.testHasValue())
+
+        Glean.pingMaker.getPingSeq("custom")
```

Comment:
Should we test that this is a non-zero value?

Reply:
Could be zero the first time...

Reply:
Maybe we should send a ping first, then we could do that.